File: apps/desktop/src/5_entities/events/model/process.ts
import type { LauncherEventPayload } from './event';

export enum ProcessPayloadType {
  Launched = 'launched',
  Finished = 'finished',
}

export interface ProcessPayload {
  instanceId: string;
  processId: string;
  event: ProcessPayloadType;
  message: string;
}

export const isProcessPayload = (
  payload: LauncherEventPayload,
): payload is ProcessPayload =>
  'event' in payload &&
  (payload.event === ProcessPayloadType.Launched ||
    payload.event === ProcessPayloadType.Finished);

File: packages/core/aether-core/src/features/events/domain/models/process.rs
use uuid::Uuid;

#[derive(Clone, Debug)]
pub struct ProcessEvent {
    pub instance_id: String,
    pub process_id: Uuid,
    pub event: ProcessEventType,
    pub message: String,
}

#[derive(Clone, Debug)]
pub enum ProcessEventType {
    Launched,
    Finished,
}
